lib/matplotlib/ticker.py

```diff
@@ -2535,8 +2535,10 @@ def __call__(self):
         tmin = ((vmin - t0) // minorstep + 1) * minorstep
         tmax = ((vmax - t0) // minorstep + 1) * minorstep
         locs = np.arange(tmin, tmax, minorstep) + t0
-        cond = np.abs((locs - t0) % majorstep) > minorstep / 10.0
-        locs = locs.compress(cond)
+        mod = np.abs((locs - t0) % majorstep)
+        cond1 = mod > minorstep / 10.0
+        cond2 = ~np.isclose(mod, np.ones_like(mod)*majorstep)
```

Comment:
`cond2 = ~np.isclose(mod, majorstep)` does not work?

Reply:
You prob need to set `atol=0` and just use `rtol` (maybe with a larger value than default).

Reply:
:+1: Yep, `atol=0` did it.